lib/decidim/spam_detection/block_spam_user_command.rb

```diff
@@ -65,9 +66,22 @@ def reason
         I18n.t("blocked_user.reason", probability: @probability)
       end
 
+      def generate_nickname
+        max_attempts = 10
+
+        max_attempts.times do
+          random_key = SecureRandom.hex(5)
+          nickname = "blocked_#{random_key}"
+          return nickname unless Decidim::User.exists?(nickname: nickname)
+        end
+
+        raise "Unable to generate a unique nickname after #{max_attempts} attempts."
```

Comment:
Prefer returning an `:invalid` or at least catch the exception raised otherwise it will break the whole job in case of collision